app/forms/declaration/CommodityMeasure.scala

```diff
@@ -28,30 +28,62 @@ import utils.validators.forms.FieldValidator._
 
 import scala.util.Try
 
-case class CommodityMeasure(supplementaryUnits: Option[String], grossMass: Option[String], netMass: Option[String])
+case class CommodityMeasure(
+  supplementaryUnits: Option[String],
+  supplementaryUnitsNotRequired: Option[Boolean],
+  grossMass: Option[String],
+  netMass: Option[String]
+)
 
 object CommodityMeasure extends DeclarationPage {
 
   implicit val format = Json.format[CommodityMeasure]
 
   val commodityFormId = "CommodityMeasure"
 
-  def applyDefault(supplementaryUnits: Option[String], grossMass: String, netMass: String): CommodityMeasure =
-    CommodityMeasure(supplementaryUnits, if (grossMass.isEmpty) None else Some(grossMass), if (netMass.isEmpty) None else Some(netMass))
+  def applyDefault(
+    supplementaryUnits: Option[String],
+    supplementaryUnitsNotRequired: Option[Boolean],
+    grossMass: String,
+    netMass: String
+  ): CommodityMeasure =
+    CommodityMeasure(
+      supplementaryUnits,
+      supplementaryUnitsNotRequired,
+      if (grossMass.isEmpty) None else Some(grossMass),
+      if (netMass.isEmpty) None else Some(netMass)
+    )
 
-  def unapplyDefault(value: CommodityMeasure): Option[(Option[String], String, String)] =
-    Some((value.supplementaryUnits, value.grossMass.getOrElse(""), value.netMass.getOrElse("")))
+  def unapplyDefault(value: CommodityMeasure): Option[(Option[String], Option[Boolean], String, String)] =
+    Some((value.supplementaryUnits, value.supplementaryUnitsNotRequired, value.grossMass.getOrElse(""), value.netMass.getOrElse("")))
 
   def applyClearance(grossMass: Option[String], netMass: Option[String]): CommodityMeasure =
-    CommodityMeasure(None, grossMass, netMass)
+    CommodityMeasure(None, None, grossMass, netMass)
 
   def unapplyClearance(value: CommodityMeasure): Option[(Option[String], Option[String])] =
     Some((value.grossMass, value.netMass))
 
   private val massFormatValidation: String => Boolean = str => validateDecimalGreaterThanZero(16)(6)(str) and containsNotOnlyZeros(str)
 
+  private val checkboxTrueValue = "true"
+
   private val mappingDefault = Forms.mapping(
     "supplementaryUnits" -> optional(text().verifying("declaration.commodityMeasure.supplementaryUnits.error", massFormatValidation)),
+    "supplementaryUnitsNotRequired" -> of(
```

Comment:
Not easy to understand what's going on here :woozy_face:

Reply:
Agree. If I may suggest, changing the second validation to:
`!(checkbox == checkboxTrueValue and nonEmpty(supUnits))`
is easier for me to understand.

Reply:
No, play form validation never is! :grin: